runtime/tr.source/trj9/z/codegen/ReduceSynchronizedFieldLoad.cpp

```diff
@@ -205,11 +250,17 @@ ReduceSynchronizedFieldLoad::perform()
 
    if (cg->getS390ProcessorInfo()->supportsArch(TR_S390ProcessorInfo::TR_z196))
       {
-      if (!cg->comp()->getOption(TR_DisableSynchronizedFieldLoad))
+      // When concurrent scavenge is enabled we need to load the object reference using a read barrier however
+      // there is no guarded load alternative for the LPD instruction. As such this optimization cannot be carried
+      // out for object reference loads under concurrent scavenge.
+      if (!TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
```

Comment:
Unrelated to this change, but we just need to disable this optimization for reference fields loads when ConcurrentScavenge is enabled.   Other primitive type loads should be okay.

Reply:
Might as well fix now. Fixed to allow primitive field loads under concurrent scavenge in 98f8017